File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/xmi/XMILoadingContext.ts
///<reference path="../Callback.ts"/>
///<reference path="../KObject.ts"/>
///<reference path="../../../../../java/util/JUArrayList.ts"/>
///<reference path="../../../../../java/util/JUHashMap.ts"/>

class XMILoadingContext {

  public xmiReader: XmlParser = null;
  public loadedRoots: KObject<any,any> = null;
  public resolvers: JUArrayList<XMIResolveCommand> = new JUArrayList<XMIResolveCommand>();
  public map: JUHashMap<string, KObject<any,any>> = new JUHashMap<string, KObject<any,any>>();
  public elementsCount: JUHashMap<string, number> = new JUHashMap<string, number>();
  public stats: JUHashMap<string, number> = new JUHashMap<string, number>();
  public oppositesAlreadySet: JUHashMap<string, boolean> = new JUHashMap<string, boolean>();
  public successCallback: Callback<Throwable> = null;

  public isOppositeAlreadySet(localRef: string, oppositeRef: string): boolean {
    return (this.oppositesAlreadySet.get(oppositeRef + "_" + localRef) != null || (this.oppositesAlreadySet.get(localRef + "_" + oppositeRef) != null));
  }

  public storeOppositeRelation(localRef: string, oppositeRef: string): void {
    this.oppositesAlreadySet.put(localRef + "_" + oppositeRef, true);
  }

}

File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/xmi/XMILoadingContext.java
package org.kevoree.modeling.api.xmi;

import org.kevoree.modeling.api.Callback;
import org.kevoree.modeling.api.KConfig;
import org.kevoree.modeling.api.KObject;
import org.kevoree.modeling.api.map.StringHashMap;

import java.util.ArrayList;

/**
 * Created by duke on 10/27/14.
 */
public class XMILoadingContext {

    public XmlParser xmiReader;

    public KObject loadedRoots = null;

    public ArrayList<XMIResolveCommand> resolvers = new ArrayList<XMIResolveCommand>();

    public StringHashMap<KObject> map = new StringHashMap<KObject>(KConfig.CACHE_INIT_SIZE,KConfig.CACHE_LOAD_FACTOR);

    public StringHashMap<Integer> elementsCount = new StringHashMap<Integer>(KConfig.CACHE_INIT_SIZE,KConfig.CACHE_LOAD_FACTOR);

    public Callback<Throwable> successCallback;

}
